pkg/gui/services/custom_commands/client.go

```diff
@@ -37,13 +41,81 @@ func NewClient(
 func (self *Client) GetCustomCommandKeybindings() ([]*types.Binding, error) {
 	bindings := []*types.Binding{}
 	for _, customCommand := range self.c.UserConfig().CustomCommands {
-		handler := self.handlerCreator.call(customCommand)
-		compoundBindings, err := self.keybindingCreator.call(customCommand, handler)
-		if err != nil {
-			return nil, err
+		if len(customCommand.SubCommands) > 0 {
+			handler := func() error {
+				return self.showCustomCommandsMenu(customCommand)
+			}
+			bindings = append(bindings, &types.Binding{
+				ViewName:    "", // custom commands menus are global; we filter the commands inside by context
```

Comment:
Why not allow optionally specifying a context on a top-level custom command menu?

Reply:
Good question; complexity? This would give us two levels of filtering by context, one for the menu itself and then again for the commands inside. I think this could be confusing, I find it easier to understand if menus are always global.

Reply:
I don't think that it would be confusing, but I also don't mind not supporting it at the moment. We can wait and see if the use case comes up.

Reply:
I'm thinking about cases like this:
```yml
customCommands:
  - context: files
    commandMenu:
      - command: whatever
        context: commits
```

This sub-command would just never show up in the menu, ever. And I think you could easily run into cases like that in practice when you start to restructure your menus and move commands from one to another.

We could probably add validation to catch and disallow such cases, but it's actually not trivial if you think about the details, and I don't think it's worth it.